kong/plugins/basic-auth/daos.lua

```diff
@@ -25,7 +25,7 @@ local SCHEMA = {
     created_at = {type = "timestamp", immutable = true, dao_insert_value = true},
     consumer_id = {type = "id", required = true, foreign = "consumers:id"},
     username = {type = "string", required = true, unique = true },
-    password = {type = "string", func = encrypt_password}
+    password = {type = "string", func = encrypt_password, trim_whitespace = false}
```

Comment:
It would be nice if your PR was updated to have two commits: 

1. Adding support (+ unit tests) for the `trim_whitespace` attribute in the schema validator
2. Adding the `trim_whitespace` attribute to the basic-auth schema (+ tests)

As per our contributing guidelines, we try to respect strict atomicity in our commits :)

Thanks!